lib/node/Creator.js

```diff
@@ -47,24 +47,28 @@ const NodeCreator = {
 
   // exponent might be null, which means there's no exponent node.
   // similarly, coefficient might be null, which means there's no coefficient
-  // the symbol node can never be null.
-  polynomialTerm (symbol, exponent, coeff, explicitCoeff=false) {
-    let polyTerm = symbol;
+  // the base node can never be null.
+  termWithCoefficient (base, exponent, coeff, explicitCoeff=false) {
```

Comment:
Thought about it for awhile and went with `Term`. I liked `TermForCollecting` at first but we might want to use these for something else in the future!